src/sketch.lisp

```diff
@@ -106,13 +115,28 @@
 
 (defmethod initialize-instance :around ((instance sketch) &key &allow-other-keys)
   (initialize-sketch)
-  (call-next-method)
+  (sdl2:in-main-thread ()
+    (call-next-method))
   (kit.sdl2:start))
 
 (defmethod initialize-instance :after ((instance sketch) &rest initargs &key &allow-other-keys)
-  (initialize-environment instance)
   (apply #'prepare instance initargs)
-  (initialize-gl instance))
+  (setf (sketch-%window instance)
+        (make-instance 'sketch-window
+                       :title (sketch-title instance)
+                       :w (sketch-width instance)
+                       :h (sketch-height instance)
+                       :fullscreen (sketch-fullscreen instance)
+                       :resizable (if (sketch-resizable instance)
```

Comment:
This argument is a CL boolean, so you should use `:resizable (sketch-resizable instance)` instead.

Reply:
This is inherited from the existing code, I didn't want to mess with it. It does seem that sdl2kit expects a CL boolean, though, so I'm happy to change it.

Reply:
It doesn't seem like it was inherited, but instead introduced in the first commit: https://github.com/Kevinpgalligan/sketch/commit/c9792e5a21eb11fc6582e4d87378c821a9a89116#diff-f109bbb16b9edc5d1c33d89883432ff91494e92ea1c331f7658b27cad82be9e8R128-R130

Reply:
I mean, in the original `define-sketch-writer` thing, it uses `sdl2-ffi:+true+`.

https://github.com/Kevinpgalligan/sketch/commit/c9792e5a21eb11fc6582e4d87378c821a9a89116#diff-f109bbb16b9edc5d1c33d89883432ff91494e92ea1c331f7658b27cad82be9e8R76

Reply:
Right, this is because it is used in a autowrapped foreign setter - `sdl2-ffi.functions:sdl-set-window-resizable`. The initarg is handled differently.